staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -518,16 +518,7 @@ func (s *serviceCache) delete(serviceName string) {
 
 // needsCleanup checks if load balancer needs to be cleaned up as indicated by finalizer.
 func needsCleanup(service *v1.Service) bool {
-	if !servicehelper.HasLBFinalizer(service) {
-		return false
-	}
-
-	if service.ObjectMeta.DeletionTimestamp != nil {
-		return true
-	}
-
-	// Service doesn't want loadBalancer but owns loadBalancer finalizer also need to be cleaned up.
-	if service.Spec.Type != v1.ServiceTypeLoadBalancer {
```

Comment:
Why remove this `if` statement?

Reply:
needsCleanup() is used in three places, used wantsloadBalancer() or needUpdate(). 
https://github.com/kubernetes/kubernetes/blob/dcfe8f5d5c9531e3196eb5036a86643c728eafe7/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L142
https://github.com/kubernetes/kubernetes/blob/dcfe8f5d5c9531e3196eb5036a86643c728eafe7/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L149
https://github.com/kubernetes/kubernetes/blob/dcfe8f5d5c9531e3196eb5036a86643c728eafe7/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L376

Since there is an appropriate check procedure for the LB type in wantsloadBalancer() and needUpdate(). , it seems that there is no need to perform additionally in needsCleanup().

Is there something I'm misunderstanding?

Reply:
/ok-to-test